server/src/models/database/socobo-user.ts

```diff
@@ -1,16 +1,64 @@
+import { IsEmail, IsNotEmpty, MinLength, ValidateIf } from "class-validator";
 import { ProviderType } from "./../enums/provider-type";
 import { Role } from "./../enums/role";
+import { ValidationGroup } from "./../enums/validation-group";
 
 export class SocoboUser {
+
   public id: number;
+
+  @ValidateIf((o) => o.email === "", {
+    groups: [ ValidationGroup.LOGIN ]
+  })
+  @IsNotEmpty({
+    groups: [ ValidationGroup.LOGIN ]
+  })
+  @MinLength(5, {
+    groups: [ ValidationGroup.LOGIN ]
+  })
   public username: string;
+
+  @ValidateIf((o) => o.username === "", {
```

Comment:
The decorators has also a message property for error messages. But I think let us merge this and improve this later.